dev/release-notes.py

```diff
@@ -0,0 +1,211 @@
+    print("## Changes since", startdate)
+
+    # Opening files with "w" resets them
+    f = open("releasenotes.md", "w")
+    f2 = open("remainingPR.md", "w")
+    f3 = open("releasenotes.json", "w")
+    jsondict = prs.copy()
+
+    prioritylist = [
+        ["kind: bug: wrong result", "Fixed bugs that could lead to incorrect results"],
+        ["kind: bug: crash", "Fixed bugs that could lead to crashes"],
+        ["kind: bug: unexpected error", "Fixed bugs that could lead to break loops"],
+        ["kind: bug", "Other fixed bugs"],
+        ["kind: enhancement", "Improved and extended functionality"],
+        ["kind: new feature", "New features"], 
+        ["kind: performance", "Performance improvements"],
+        ["topic:libgap", "Improvements in the experimental way to allow 3rd party code to link GAP as a library"],
+        ["topic: julia", "Improvements in the experimental support for using the **Julia** garbage collector"],
+        ["topic: documentation", "Changed documentation"],
+        ["topic: packages", "Packages"]
+    ]
+
+    # TODO: why does this need a special treatment? 
+    # Adding it to the prioritylist could ensure that it goes first
+    f.write("## Release Notes \n\n")
+    f.write("### " + "New features and major changes" + "\n")
+    removelist = []
+    for k in prs:
+        # The format of an entry of list is: ["title of PR", "Link" (Alternative the PR number can be used), [ list of labels ] ]
+        if "release notes: highlight" in prs[k]["labels"]:
+            # TODO: writing up these details should be a function
+            f.write("- [#")
+            issuenumber = str(k)
+            f.write(issuenumber)
+            f.write("](")
+            f.write("https://github.com/gap-system/gap/pull/" + str(k))
+            f.write(") ")
+            f.write(prs[k]["title"])
+            f.write("\n")
+            removelist.append(k)
+    for item in removelist:
+        del prs[item]
+    f.write("\n")
+
+
+    removelist = []
+    for k in prs:
+        if "release notes: not needed" in prs[k]["labels"]:
+            removelist.append(k)
+    for item in removelist:
+        del prs[item]
+        del jsondict[item]
+
+
+    f2.write("### " + "release notes: to be added" + "\n")
+    removelist = []
+    for k in prs:
+        if "release notes: to be added" in prs[k]["labels"]:
+            f2.write("- [#")
+            issuenumber = str(k)
+            f2.write(issuenumber)
+            f2.write("](")
+            f2.write("https://github.com/gap-system/gap/pull/" + str(k))
+            f2.write(") ")
+            f2.write(prs[k]["title"])
+            f2.write("\n")
+            removelist.append(k)
+    for item in removelist:
+        del prs[item]
+    f2.write("\n")
+
+
+    f2.write("### Uncategorized PR" + "\n")
+    removelist = []
+    for k in prs:
+        #if not "release notes: use title" in item[2]:
+        if not "release notes: added" in prs[k]["labels"]:
```

Comment:
For the next release notes we have to use "release notes: use title" instead of "release notes: added" at this point. At the moment we use "release notes: added" to see whether the script works with the current release notes.